### ISSUE TITLE
pydata-sphinx-theme 0.13 causes doc builds to fail

### ISSUE THREAD

**Issue by oscargus** [matplotlib/matplotlib#25344]
### Summary

Recent doc failures:

```
WARNING: `footer_items` is deprecated. Use `footer_start` or `footer_end` instead.
WARNING: Path to light image logo does not exist: images/logo2.svg
WARNING: Path to dark image logo does not exist: images/logo_dark.svg
```

Difference seems to be pydata-sphinx-theme 0.13 (can be more packages later on, but this is the first I saw).

### Proposed fix

Easiest is probably to pin to 0.12, better to fix the underlying issues.

**Comment by oscargus** [comment]
I guess this is also what kills the AppVeyor tests now?

https://ci.appveyor.com/project/matplotlib/matplotlib/builds/46452127/job/unu8todaql8911jj

```
AILED lib/matplotlib/tests/test_doc.py::test_sphinx_gallery_example_header - AssertionError: assert '\n.. DO NOT ..._glr_{1}:\n\n' == '\n.. DO NOT ..._glr_{1}:\n\n'
  Skipping 242 identical leading characters in diff, use -v to show
    
    
  -         Click :ref:`here <sphx_glr_download_{1}>`
  ?        ------         ^
  +         :ref:`Go to the end <sphx_glr_download_{1}>`
  ?               +++++++  ^ ++
            to download the full example code{2}
    
    .. rst-class:: sphx-glr-example-title
    
    .. _sphx_glr_{1}:
```

**Comment by QuLogic** [comment]
This should have been fixed by #25423 and #25429.
